buildSrc/src/main/kotlin/TargetsConfig.kt

```diff
@@ -0,0 +1,178 @@
+
+    val files = project.projectDir.listFiles() ?: emptyArray()
+
+    val hasCommon = files.any { it.name == "common" }
+    val hasJvmAndNix = hasCommon || files.any { it.name == "jvmAndNix" }
+    val hasPosix = hasCommon || files.any { it.name == "posix" }
+    val hasDesktop = hasPosix || files.any { it.name == "desktop" }
+    val hasNix = hasPosix || files.any { it.name == "nix" }
+    val hasDarwin = hasNix || files.any { it.name == "darwin" }
+    val hasJs = hasCommon || files.any { it.name == "js" }
+    val hasJvm = hasCommon || files.any { it.name == "jvm" }
+
+    kotlin {
+        if (hasJvm) {
+            jvm()
+            configureJvm()
+        }
+
+        if (hasJs) {
+            js {
+                nodejs()
+                browser()
+            }
+
+            configureJs()
+        }
+
+        if (hasPosix) {
+            posixTargets()
+            extra.set("hasNative", true)
+        }
+
+        if (hasDarwin) extra.set("hasNative", true)
+
+        if (IDEA_ACTIVE) {
+            if (hasPosix) createIdeaTarget("posix")
+            if (hasNix) createIdeaTarget("nix")
+            if (hasDarwin) createIdeaTarget("darwin")
+            if (hasDesktop) createIdeaTarget("desktop")
+        } else {
+            sourceSets {
+                if (hasPosix) {
+                    val posixMain by creating
+                    val posixTest by creating
+                }
+                if (hasNix) {
+                    val nixMain by creating
+                    val nixTest by creating
+                }
+                if (hasDarwin) {
+                    val darwinMain by creating
+                    val darwinTest by creating
+                }
+                if (hasDesktop) {
+                    val desktopMain by creating
+                    val desktopTest by creating
+                }
+            }
+        }
+
+        sourceSets {
+            if (hasCommon) {
+                val commonMain by getting {
+                    dependencies {
+                        api("org.jetbrains.kotlin:kotlin-stdlib-common")
+                        api("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version")
+                    }
+                }
+                val commonTest by getting {
+                    dependencies {
+                        api("org.jetbrains.kotlin:kotlin-test-common:$kotlin_version")
+                        api("org.jetbrains.kotlin:kotlin-test-annotations-common:$kotlin_version")
+                    }
+                }
+
+                extra.set("commonStructure", true)
+            }
+            if (hasJvmAndNix) {
+                val jvmAndNixMain by creating {
+                    findByName("commonMain")?.let { dependsOn(it) }
+
+                    dependencies {
+                        api("org.jetbrains.kotlin:kotlin-stdlib-common")
+                        api("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version")
```

Comment:
is it needed, as `common` already contains it?

Reply:
Thanks, fixed